capstone-rs/src/instruction.rs

```diff
@@ -385,6 +388,44 @@ impl<'a> Display for Instructions<'a> {
         Ok(())
     }
 }
+/// Registers read and/or written to by a given instruction
+#[derive(Debug)]
+pub struct RegAccess {
+    /// `Vec` containing the `RegId` of the registers read from
+    read: Vec<RegId>,
+     
+    /// `Vec` containing the `RegId` of the registers written to
+    write: Vec<RegId>,
+}
+impl RegAccess {
+    /// Converts the two slices of u16's from the API into the `RegAccss` type
+    pub(crate) fn from_slice_u16(
+        registers_read: &CsRegs,
+        registers_read_count: usize,
+        registers_written: &CsRegs,
+        registers_written_count: usize,
+        ) -> Self {
+            let read = registers_read[0..registers_read_count]
+                .iter()
+                .cloned()
+                .map(|x| RegId::from(x as u32))
```

Comment:
since `RegIdInt` is defined to be `u16`, could you define a new impl `From<u16>` for `RegId`? There's no need to cast here.

You could already just do `RegId(x)` (since the field is public).